### PR TITLE
Test strategies and ExcludeFromCodeCoverage

### PR DESCRIPTION
**Pull request by Arthuresm**
This functionality allows you to start the template according to the requirements necessary to Quality Gate.

### CODE REVIEW

File: content/Api/Blip.Api.Template.Facades/Blip.Api.Template.Facades.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netstandard2.1</TargetFramework>
-    <ProjectGuid>sonarqube-guid-facades</ProjectGuid>
+    <ProjectGuid>{C4E5B837-8C8D-47ED-973E-FB70664FFB60}</ProjectGuid>
```
Comment (lfmundim) [inline content/Api/Blip.Api.Template.Facades/Blip.Api.Template.Facades.csproj:5]: whenever you build the project, or open in VS even, the placeholder gets replaced by the generated GUID. Revert back to the placeholder

File: content/Api/Blip.Api.Template.Models/Blip.Api.Template.Models.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netstandard2.1</TargetFramework>
-    <ProjectGuid>sonarqube-guid-models</ProjectGuid>
+    <ProjectGuid>{46EC828B-835E-4475-B0ED-F1EFC0DBCB51}</ProjectGuid>
```
Comment (lfmundim) [inline content/Api/Blip.Api.Template.Models/Blip.Api.Template.Models.csproj:5]: whenever you build the project, or open in VS even, the placeholder gets replaced by the generated GUID. Revert back to the placeholder

File: content/Api/Blip.Api.Template.Services/Blip.Api.Template.Services.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netstandard2.1</TargetFramework>
-    <ProjectGuid>sonarqube-guid-services</ProjectGuid>
+    <ProjectGuid>{8562C21A-DC12-4F70-AA93-C136563F2CD8}</ProjectGuid>
```
Comment (lfmundim) [inline content/Api/Blip.Api.Template.Services/Blip.Api.Template.Services.csproj:5]: whenever you build the project, or open in VS even, the placeholder gets replaced by the generated GUID. Revert back to the placeholder

File: content/Api/Blip.Api.Template.Tests/Blip.Api.Template.Tests.csproj [hunk 4]
```diff
@@ -1,18 +1,21 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>netcoreapp3.1</TargetFramework>
-    <ProjectGuid>sonarqube-guid-tests</ProjectGuid>
+    <ProjectGuid>{FE7A5C89-B282-4526-A91B-45A586FA9BD7}</ProjectGuid>
```
Comment (lfmundim) [inline content/Api/Blip.Api.Template.Tests/Blip.Api.Template.Tests.csproj:5]: whenever you build the project, or open in VS even, the placeholder gets replaced by the generated GUID. Revert back to the placeholder

File: content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/ApiExceptionHandlingStrategyTests.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
+﻿using System.Threading.Tasks;
+
+using Blip.Api.Template.Facades.Strategies.ExceptionHandlingStrategies;
+using Blip.Api.Template.Tests.Setup.Controller;
+
+using FluentAssertions;
```
Comment (lfmundim) [inline content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/ApiExceptionHandlingStrategyTests.cs:6]: the entire project is using `Shouldly` to test. Why change it now?

File: content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/ApiExceptionHandlingStrategyTests.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            var apiExceptionHandlingStrategy = CreateApiExceptionHandlingStrategy();
+
+            var context = ControllerSetup.HttpContext;
+            var exception = ControllerSetup.GetApiException(Faker);
+
+            var result = await apiExceptionHandlingStrategy.HandleAsync(
+                context,
+                exception);
+
+            result.Response.StatusCode.Should().Be((int)exception.StatusCode);
```
Comment (lfmundim) [inline content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/ApiExceptionHandlingStrategyTests.cs:42]: if using `Shouldly` just use `ShouldBe((int)...)`

File: content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/NotImplementedExceptionHandlingStrategyTests.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
+﻿using System.Threading.Tasks;
+
+using Blip.Api.Template.Facades.Strategies.ExceptionHandlingStrategies;
+using Blip.Api.Template.Tests.Setup.Controller;
+
+using FluentAssertions;
```
Comment (lfmundim) [inline content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/NotImplementedExceptionHandlingStrategyTests.cs:6]: again, why change from Shouldly?

File: content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/NotImplementedExceptionHandlingStrategyTests.cs [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            var notImplementedExceptionHandlingStrategy = CreateNotImplementedExceptionHandlingStrategy();
+
+            var context = ControllerSetup.HttpContext;
+            var exception = ControllerSetup.GetNotImplementedException(Faker);
+
+            var result = await notImplementedExceptionHandlingStrategy.HandleAsync(
+                context,
+                exception);
+
+            result.Response.StatusCode.Should().Be(StatusCodes.Status501NotImplemented);
```
Comment (lfmundim) [inline content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/NotImplementedExceptionHandlingStrategyTests.cs:44]: same comment as on other test file

File: content/Api/Blip.Api.Template/Blip.Api.Template.csproj [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netcoreapp3.1</TargetFramework>
-    <ProjectGuid>sonarqube-guid-ui</ProjectGuid>
+    <ProjectGuid>{2E3DD875-4CFB-41EF-B627-7BA155C63F93}</ProjectGuid>
```
Comment (lfmundim) [inline content/Api/Blip.Api.Template/Blip.Api.Template.csproj:5]: whenever you build the project, or open in VS even, the placeholder gets replaced by the generated GUID. Revert back to the placeholder

**Review by lfmundim — CHANGES_REQUESTED** [review]
It is very important that `.csproj` guid placeholders _stay as placeholders_. Everything breaks if they don't. 

Why start using `FluentAssertions` if the entire project uses `Shouldly`?

File: content/Api/Blip.Api.Template.Facades/Blip.Api.Template.Facades.csproj (hunk 1 above)
Comment (Arthuresm) [reply content/Api/Blip.Api.Template.Facades/Blip.Api.Template.Facades.csproj:5]: I forgot that point. Thanks for remember!

File: content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/ApiExceptionHandlingStrategyTests.cs (hunk 5 above)
Comment (Arthuresm) [reply content/Api/Blip.Api.Template.Tests/Strategies/ExceptionHandlingStrategies/ApiExceptionHandlingStrategyTests.cs:6]: Just a preference. I will return to `Shouldly `to keep the same package already used.

**Review by victor-smok — APPROVED** [review]

**Review by lfmundim — APPROVED** [review]

**Comment by lfmundim** [comment]
One final note, @Arthuresm 

edit the [`.nuspec`](https://github.com/lfmundim/Blip.Api.Template.K8s/blob/master/Blip.Api.Template.K8s.nuspec) file to version `1.8.2` and add your release notes, so I can publish the package to NuGet

**Comment by Arthuresm** [comment]
> One final note, @Arthuresm
> 
> edit the [`.nuspec`](https://github.com/lfmundim/Blip.Api.Template.K8s/blob/master/Blip.Api.Template.K8s.nuspec) file to version `1.8.2` and add your release notes, so I can publish the package to NuGet

**Review by lfmundim — APPROVED** [review]

**Pull request merged**